ruby/bps-stan/lib/bps/subscriber/stan.rb

```diff
@@ -0,0 +1,41 @@
+require 'cgi'
+require 'stan/client'
+
+module BPS
+  module Subscriber
+    class STAN < Abstract
+      def initialize(cluster_id, client_id, nats: {}, **opts)
+        super()
+
+        # handle TLS if CA file is provided:
+        if !nats[:tls] && nats[:tls_ca_file]
+          ctx = OpenSSL::SSL::SSLContext.new
+          ctx.set_params
+          ctx.ca_file = nats.delete(:tls_ca_file)
+          nats[:tls] = ctx
+        end
+
+        @client = ::STAN::Client.new
+        @client.connect(cluster_id, client_id, nats: nats, **opts.slice(*::BPS::STAN::Utils::CLIENT_OPTS.keys))
```

Comment:
Maybe entire connection creating could be moved to smth like `@client = ...::Utils.connect(...)` or so? Right now it's just copy-paste.

Reply:
yeah, would be nice, not sure if you need extra `Utils::` namespace either, just:

```
::BPS::STAN.connect # => ::STAN::Client
::BPS::STAN::CLIENT_OPTS # => { ... }
::BPS::STAN.coercer # => ...
```